src/main/java/com/dataartisans/flink/dataflow/translation/wrappers/streaming/FlinkAbstractParDoWrapper.java

```diff
@@ -0,0 +1,259 @@
+		}
+
+		@Override
+		public <T> void sideOutputWithTimestamp(TupleTag<T> tag, T output, Instant timestamp) {
+			sideOutputWithTimestampHelper(element, output, timestamp, collector, tag);
+		}
+
+		@Override
+		protected <AggInputT, AggOutputT> Aggregator<AggInputT, AggOutputT> createAggregatorInternal(String name, Combine.CombineFn<AggInputT, ?, AggOutputT> combiner) {
+			Accumulator acc = getRuntimeContext().getAccumulator(name);
+			if (acc != null) {
+				AccumulatorHelper.compareAccumulatorTypes(name,
+						SerializableFnAggregatorWrapper.class, acc.getClass());
+				return (Aggregator<AggInputT, AggOutputT>) acc;
+			}
+
+			SerializableFnAggregatorWrapper<AggInputT, AggOutputT> accumulator =
+					new SerializableFnAggregatorWrapper<>(combiner);
+			getRuntimeContext().addAccumulator(name, accumulator);
+			return accumulator;
+		}
+	}
+
+	protected void checkTimestamp(WindowedValue<IN> ref, Instant timestamp) {
+		if (timestamp.isBefore(ref.getTimestamp().minus(doFn.getAllowedTimestampSkew()))) {
+			throw new IllegalArgumentException(String.format(
+					"Cannot output with timestamp %s. Output timestamps must be no earlier than the "
+							+ "timestamp of the current input (%s) minus the allowed skew (%s). See the "
+							+ "DoFn#getAllowedTimestmapSkew() Javadoc for details on changing the allowed skew.",
+					timestamp, ref.getTimestamp(),
+					PeriodFormat.getDefault().print(doFn.getAllowedTimestampSkew().toPeriod())));
+		}
+	}
+
+	protected <T> WindowedValue<T> makeWindowedValue(
+			T output, Instant timestamp, Collection<? extends BoundedWindow> windows, PaneInfo pane) {
+		final Instant inputTimestamp = timestamp;
+		final WindowFn windowFn = windowingStrategy.getWindowFn();
+
+		if (timestamp == null) {
+			timestamp = BoundedWindow.TIMESTAMP_MIN_VALUE;
+		}
+
+		if (windows == null) {
+			try {
+				windows = windowFn.assignWindows(windowFn.new AssignContext() {
+					@Override
+					public Object element() {
+						throw new UnsupportedOperationException(
+								"WindowFn attempted to access input element when none was available"); // TODO: 12/16/15 aljoscha's comment in slack
+					}
+
+					@Override
+					public Instant timestamp() {
+						if (inputTimestamp == null) {
+							throw new UnsupportedOperationException(
+									"WindowFn attempted to access input timestamp when none was available");
+						}
+						return inputTimestamp;
+					}
+
+					@Override
+					public Collection<? extends BoundedWindow> windows() {
+						throw new UnsupportedOperationException(
+								"WindowFn attempted to access input windows when none were available");
+					}
+				});
+			} catch (Exception e) {
+				Throwables.propagateIfInstanceOf(e, UserCodeException.class);
+				throw new UserCodeException(e);
+			}
+		}
+
+		return WindowedValue.of(output, timestamp, windows, pane);
+	}
+
+	///////////			ABSTRACT METHODS TO BE IMPLEMENTED BY SUBCLASSES			/////////////////
```

Comment:
I think this comment makes the code clearer to someone who reads it, as these are the methods to be implemented by subclasses.